P/PMIx/build_tarballs.jl

```diff
@@ -7,7 +7,7 @@ version = v"4.1.0"
 
 # Collection of sources required to complete build
 sources = [
-    ArchiveSource("https://github.com/pmix/pmix/releases/download/v$(version)/pmix-$(version).tar.bz2", "145f05a6c621bfb3fc434776b615d7e6d53260cc9ba340a01f55b383e07c842e")
```

Comment:
This rebuilds the package again even if nothing changes, maybe do it in a separate PR with `[skip ci]`?